src/interpreter.ts

```diff
@@ -1038,10 +1038,14 @@ export class Interpreter {
             }
             case "sha256": {
                 ensureFunArity(1, ast.args, ast.loc);
-                const str = ensureString(
-                    this.interpretExpression(ast.args[0]!),
-                    ast.args[0]!.loc,
-                );
+                const expr = this.interpretExpression(ast.args[0]!);
+                if (expr instanceof Slice) {
+                    throwNonFatalErrorConstEval(
```

Comment:
Since this returns a non fatal error, it means that if someone actually passes a Slice as argument to `sha256`, the interpreter will not evaluate the expression, but the code generation will still accept the expression. For example, this:
```
let a = sha256(s);   // where s has type Slice       
``` 
will be accepted by the compiler, but will not be simplified by the interpreter.

Is this the expected behavior? Is it possible to add a positive test where `sha256` receives a Slice?

Reply:
Yup, now that's expected. And there's such test in `intrinsics.spec.ts`